invenio_accounts/models.py

```diff
@@ -463,3 +468,153 @@ def delete_by_user(cls, method, user):
         """Unlink a user from an external id."""
         with db.session.begin_nested():
             cls.query.filter_by(id_user=user.id, method=method).delete()
+
+
+class DomainOrg(db.Model):
+    """Domain organisation."""
+
+    __tablename__ = "accounts_domain_org"
+
+    id = db.Column(db.Integer(), primary_key=True, autoincrement=True)
+
+    pid = db.Column(db.String(255), unique=True, nullable=True)
+    """Persistent identifier for organisation."""
+
+    name = db.Column(db.String(255), nullable=False)
+    """Name of organisation."""
+
+    json = db.Column(
+        json_field,
+        default=lambda: dict(),
+        nullable=False,
+    )
+    """Store additional metadata about the organisation."""
+
+    parent_id = db.Column(
+        db.Integer(), db.ForeignKey("accounts_domain_org.id"), nullable=True
+    )
+    """Link to parent organisation."""
+
+    parent = db.relationship("DomainOrg", remote_side=[id])
+    """Relationship to parent."""
+
+    domains = db.relationship("Domain", back_populates="org")
+    """Relationship to domains for this organisation."""
+
+    @classmethod
+    def create(cls, pid, name, json=None, parent=None):
+        """Create a domain organisation."""
+        obj = cls(pid=pid, name=name, json=json or {}, parent=parent)
+        db.session.add(obj)
+        return obj
+
+
+class DomainCategory(db.Model):
```

Comment:
Label a domain with a category, on Zenodo will be used to classify domains in e.g. spammer, mail-provider, org, company.